cloud-common/src/main/java/com/sequenceiq/cloudbreak/common/service/token/CachedRemoteTokenService.java

```diff
@@ -70,8 +70,8 @@ public OAuth2Authentication loadAuthentication(String accessToken) throws Authen
         }
         Jwt jwtToken = JwtHelper.decode(accessToken);
         try {
-            Map<String, Object> claims = objectMapper.readValue(jwtToken.getClaims(), new MapTypeReference());
-            return claims.get("user") == null ? getOAuth2Authentication(accessToken) : getSSOAuthentication(accessToken);
+            Map<String, String> claims = objectMapper.readValue(jwtToken.getClaims(), new MapTypeReference());
+            return "uaa".equals(claims.get("zid")) ? getOAuth2Authentication(accessToken) : getSSOAuthentication(accessToken);
```

Comment:
Probably the "zid" value will be "uaa" in our cases, but if a customer configures the UAA differently with multiple zones then the "zid" has different value. Of course if the "zid" means the zone id in this case.
I have tested it with LDAP configured, so it shouldn't cause any issues on our envs.